### DISCUSSION TITLE
How to change lazygit colors

### DISCUSSION THREAD

**Discussion by EricDriussi** [LazyVim/LazyVim#4251] [Q&A] [answered]
Is there a way to change specific colors within lazygit?

I understand it's managed based on the selected colorscheme but using [gruvbox-material](https://github.com/sainnhe/gruvbox-material) for example the background color seems to be the same as the background color for the selected file/line.

The only way to tell what I have selected is by looking at which line is bold. How can I change the background of the selected item?

![image](https://github.com/user-attachments/assets/6aeae1eb-dba7-41aa-9774-ef344ee62dcd)

**Comment by dpetka2001** [comment]
You can try setting the background color for `Visual` highlight group to something that you're able to distinguish easily (since Lazyvim uses `Visual` from whatever colorscheme it is set to highlight the selected line). For example after you open Neovim just do `:hi Visual guibg=#ffffff` and this will set the background color to white. Then when you open Lazygit the selected line should have a white background color.

**Reply by EricDriussi** [reply]
Thanks, but I was hoping to find a way to be a bit more specific.

I like the colorscheme as is, It's just this particular instance I was hoping to change.

Maybe it's easier to change the whole lazygit background color? Is that grabbing the background from the Pmenu hl group or something like that?

**Reply by dpetka2001** [reply]
It's grabbing the background from whatever the `Visual` highlight group is set to in the colorscheme that someone uses.

**Reply by EricDriussi** [reply]
Sorry, I understand that the Visual hl group dictates the background for the selected item.
I'm wondering what hl group it looks for to set the background for the whole lazygit panel, since changing that might (or might not) be less disruptive to the overall colorscheme.

**Reply by dpetka2001** [reply, marked as answer]
You can see the highlight groups that LazyVim sets for `lazygit` [here](https://github.com/LazyVim/LazyVim/blob/12818a6cb499456f4903c5d8e68af43753ebc869/lua/lazyvim/util/lazygit.lua#L22-L33). But I don't think it sets a background color for the lazygit panel. The background color seems to be inherited from the user's colorscheme itself. What you could do is set `vim.g.lazygit_config = false` in your `options.lua` file and then copy the values from `~/.cache/nvim/lazygit-theme.yml` to your `~/.config/lazygit/config.yml` and change the values that you want to the colors you like. This won't change anything in your colorscheme, it will just use static colors for your lazygit config, but you lose the advantage of changing colorschemes and lazygit updating the colors automatically.

**Comment by linkarzu** [comment]
> You can see the highlight groups that LazyVim sets for lazygit [here](https://github.com/LazyVim/LazyVim/blob/12818a6cb499456f4903c5d8e68af43753ebc869/lua/lazyvim/util/lazygit.lua#L22-L33). But I don't think it sets a background color for the lazygit panel. The background color seems to be inherited from the user's colorscheme itself. What you could do is set vim.g.lazygit_config = false in your options.lua file and then copy the values from ~/.cache/nvim/lazygit-theme.yml to your ~/.config/lazygit/config.yml and change the values that you want to the colors you like. This won't change anything in your colorscheme, it will just use static colors for your lazygit config, but you lose the advantage of changing colorschemes and lazygit updating the colors automatically.

Appreciate it @dpetka2001 as always, you answering my lazyvim questions, thank you! 

Is there a way for me to change the highlight group that `selectedLineBgColor` is mapped to? Instead of `Visual` I would like it to be `CursorLine`, in the commit you shared above, I see this in the lazy.lua file, but that lazy.lua file is not available anymore in the main branch
```
  selectedLineBgColor = { bg = "Visual" }, -- set to `default` to have no background colour
```

I found this [answer](https://github.com/LazyVim/LazyVim/issues/2846#issuecomment-2022401716) from folke and trying to configure that highlight group where I customize the rest of my highlight groups. After setting it up, I do see it when I run the `:highlight` command (I have `selectedLineBgColorxxx` 2 times because I was testing)
```
selectedLineBgColorxxx guibg=#314154
@lazygitxxx cleared
@lazygit.theme.selectedLineBgColorxxx guibg=#314154
```

So is the only way to set this up now by setting the `vim.g.lazygit_config = false` as you described above? It would be much easier for me to just change this highlight group and that's it.

**Reply by dpetka2001** [reply]
If you're using Lazyvim prior to 13 then yes. With the release of 13 that setting got moved into Snacks. I haven't messed around with it yet regarding the colorscheme, so i can't give you a definite answer as i might be wrong.

**Reply by folke** [reply]
You can change the theme. See the general snacks docs and https://github.com/folke/snacks.nvim/blob/main/docs/lazygit.md

**Reply by linkarzu** [reply]
I'll look into snacks because I'm on 13, thanks for pointing me in the right direction

**Comment by linkarzu** [comment]
To add more context, this is why I want to change it, my colorscheme does not favor the white fg
<img src="https://github.com/user-attachments/assets/7e3eb5d3-7482-4b43-a0f9-281aeaf090a7" alt="linkarzu com 2024-11-09 at 09 00 10" width="300"/>

**Reply by folke** [reply]
You can change the theme. See the general snacks docs and https://github.com/folke/snacks.nvim/blob/main/docs/lazygit.md

**Reply by linkarzu** [reply]
Appreciate it!
